app/jobs/hud_reports/store_artifacts_and_cleanup_job.rb

```diff
@@ -0,0 +1,143 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+# Background job to store HUD report artifacts in S3 and immediately cleanup RDS data
+# This job runs after a report is completed to archive intermediate data and free up RDS space
+module HudReports
+  class StoreArtifactsAndCleanupJob < BaseJob
+    queue_as ENV.fetch('DJ_LONG_QUEUE_NAME', :long_running)
+
+    def perform(report_id)
+      report = HudReports::ReportInstance.find(report_id)
+
+      Rails.logger.info "Processing report #{report_id} for S3 storage and RDS cleanup"
+
+      begin
+        # Step 1: Store artifacts to S3
+        store_artifacts_to_s3(report)
+
+        # Step 2: Update report status (mark as artifacts stored)
+        update_report_status(report)
+
+        # Step 3: Clean up RDS data
+        cleanup_rds_data(report)
+
+        Rails.logger.info "Successfully processed report #{report_id}: stored to S3 and cleaned RDS"
+      end
```

Comment:
Did you intend to add a rescue here?
Also, it looks like if the artifacts fail to get stored, it'll still make them as stored, and cleanup the RDS.

`store_artifacts_to_s3` should probably return a true or false, and if false, it should raise and fail catastrophically.

Additionally, we should probably have some sort of wrapper around `cleanup_rds_data` so that if it fails we know we need to investigate and try again.  Maybe it could happen in a transaction?  Not sure if that's possible given the size of the changes...